src/realm/object-store/c_api/schema.cpp

```diff
@@ -55,6 +55,28 @@ RLM_API uint64_t realm_get_schema_version(const realm_t* realm)
     return rlm->schema_version();
 }
 
+RLM_API uint64_t realm_get_persisted_schema_version(const realm_config_t* config)
+{
+
+    auto conf = RealmConfig();
+    conf.schema_version = ObjectStore::NotVersioned;
+    conf.path = config->path;
+    conf.encryption_key = config->encryption_key;
+
+    if (config->sync_config) {
+        conf.sync_config = nullptr;
+        conf.force_sync_history = true;
+    }
+
+    return wrap_err([&]() {
+        auto realm = new shared_realm{Realm::get_shared_realm(conf)};
```

Comment:
There's no reason to wrap this in an extra `shared_realm` since it isn't returned to the caller.